x/sync/syncworkheap.go

```diff
@@ -35,7 +37,7 @@ type syncWorkHeap struct {
 
 func newSyncWorkHeap() *syncWorkHeap {
 	return &syncWorkHeap{
-		priorityHeap: make([]*heapItem, 0),
+		innerHeap: make([]*heapItem, 0),
```

Comment:
I know this isn't part of this PR - but I don't think we need to be doing `make([]T, 0)` here. The default `nil` should be fine.